### PR TITLE
Add serde derive macros where applicable

### PR DESCRIPTION
**Pull request by daviessm**
I had a requirement to serialize packets. What would you think of adding `serde` support to the package, behind a feature flag?

### CODE REVIEW

**Comment by JulianSchmid** [comment]
Hi @daviessm ,

I like it, but I would as you proposed put it behind a non default feature flag.

Greets
Julian

File: src/transport/tcp.rs [hunk 1]
```diff
@@ -70,9 +71,16 @@ pub struct TcpHeader {
     ///the URG control bit set.
     pub urgent_pointer: u16,
     ///Buffer containing the options of the header (note that the data_offset defines the actual length). Use the options() method if you want to get a slice that has the actual length of the options.
+    #[cfg_attr(feature = "serde", serde(skip), serde(default = "default_options_buffer"))]
```
Comment (JulianSchmid) [inline src/transport/tcp.rs:74]: I think a manual implementation of `Serialize` & `Deserialize` is needed for `TcpHeader` as well. Only the parts of the `options_buffer` that is actually filled should be serialized and/or deserialized. The length of actually filled data in the options_buffer is defined by the `_data_offset` field. The `_data_offset` field itself can be skipped as it is implicitly encoded in the length of the `options_buffer`.

We could also think about instead serializing the decoded options if that is possible. But that is maybe overkill.

File: src/internet/ipv4.rs [hunk 2]
```diff
@@ -28,6 +29,7 @@ pub struct Ipv4Header {
     pub destination: [u8;4],
     /// Length of the options in the options_buffer in bytes.
     options_len: u8,
+    #[cfg_attr(feature = "serde", serde(skip), serde(default = "default_options_buffer"))]
     options_buffer: [u8;40]
```
Comment (JulianSchmid) [inline src/internet/ipv4.rs:33]: Same as with the `TcpHeader` & `Ipv6RawExtension` I would implement `Serialize` & `Deserialize` manually as options buffer is not required to be fully filled (see comments there).

**Review by JulianSchmid — CHANGES_REQUESTED** [review]

**Comment by JulianSchmid** [comment]
Sorry I forgot: Thanks for the pull request 👍 .

**Comment by daviessm** [comment]
Hi @JulianSchmid, thanks for the review! I implemented this because I thought I needed it in a project but ended up writing a custom serializer in my code instead so I won't have any time to work on this any more. I'd be happy for this to be closed or taken over by someone else, whatever works best for you.

**Comment by JulianSchmid** [comment]
Hi @daviessm , then I would take over the PR and implement it.

**Comment by JulianSchmid** [comment]
I moved to branch to this repo and opened up a new PR https://github.com/JulianSchmid/etherparse/pull/45

**Pull request merged**